.github/workflows/ci.yml

```diff
@@ -43,40 +43,53 @@ jobs:
       - name: basedpyright
         run: uv run poe pyright
 
-  typetest:
-    timeout-minutes: 5
+  generate-matrix:
+    name: Generate Test Matrix
     runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
-        uses: astral-sh/setup-uv@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
 
-      - name: install
-        run: uv sync --frozen --python 3.13
-
-      - name: typetest
-        run: uv run poe typetest
+      - name: Install Python Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install scipy
 
-      - name: stubtest
-        run: uv run poe stubtest
+      - name: Generate Matrix
+        id: set-matrix
+        shell: bash
+        run: |
+          matrix=$(python scripts/generate_matrix.py | jq -c)
+          echo "matrix=$matrix" >> $GITHUB_OUTPUT
 
-  typetest-oldest-supported-numpy:
-    timeout-minutes: 5
+  typetest:
+    needs: generate-matrix
+    timeout-minutes: 20
     runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJSON(needs.generate-matrix.outputs.matrix) }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
```

Comment:
Maybe I'm missing something, but I thought this wasn't needed anymore because of `uv` 🤔

Reply:
You're right... force of habit :P
I'm not used to using uv yet

Reply:
> I'm not used to using uv yet

and given that the CI wasn't actually testing the oldest numpy versions before today, apparently neither am I 😛